### PR TITLE
File io

### PR DESCRIPTION
**Pull request by td-extreme**

### CODE REVIEW

File: src/httpserver/file_io/file_input.clj [hunk 1]
```diff
@@ -0,0 +1,17 @@
+(ns httpserver.file-io.file-input
+  (:require [clojure.java.io :as io])
+  (:import java.io.FileInputStream))
+
+(defn- partial-content [array-of-bytes start end]
+  (subvec (mapv byte array-of-bytes) start end))
+
```
Comment (td-extreme) [inline src/httpserver/file_io/file_input.clj:7]: As I have been working on the partial content handler. I realize this should probably not be in the `file-input` ns and instead should be in the `partial-content` handler ns.

File: spec/httpserver/file_io/file_input_spec.clj [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+           [httpserver.file-io.file-input :as file-input]))
+
+(describe "file io"
+  (describe "reading a file"
+    (it "returns contents of testfile as byte-array"
+        (should= (type (byte-array 3)) (type (file-input/get-file-contents-as-bytes
+                              "./folder_for_io_testing"
+                              "/test.txt"))))
+   (it "returns a byte-array of length 32 when asked to open test.txt file"
+     (should= 33 (count (file-input/get-file-contents-as-bytes
```
Comment (mikeknep) [inline spec/httpserver/file_io/file_input_spec.clj:12]: Test expects 33, description expects 32.

File: src/httpserver/file_io/file_input.clj (hunk 1 above)
Comment (mikeknep) [inline src/httpserver/file_io/file_input.clj:7]: Does this comment above apply to all functions in this namespace? Just clarifying, as I'm not sure what this specific ns would contain / be used for.

Comment (td-extreme) [inline src/httpserver/file_io/file_input.clj:7]: It applies to the function above `partial-content` and the method `get-file-partial -content`.  It does not apply to the function `get-file-contents-as-bytes` 

This ns would just be for getting the contents of a file. The entire contents of a file.

**Pull request merged**